larpandoracontent/LArHelpers/LArGeometryHelper.cc

```diff
@@ -422,6 +422,58 @@ bool LArGeometryHelper::IsInGap3D(const Pandora &pandora, const CartesianVector
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
+float LArGeometryHelper::ProjectAcrossGap3D(const Pandora &pandora, const CartesianVector &testPoint, const CartesianVector &testDirection,
+        const float gapTolerance, const int recurseLimit)
+{
+    float bestGapSize = std::numeric_limits<float>::max();
+
+    for (const DetectorGap *const detectorGap : pandora.GetGeometry()->GetDetectorGapList())
+    {
+        if (!detectorGap->IsInGap(testPoint, TPC_3D, gapTolerance))
+            continue;
+
+        const LineGap* lineGap = dynamic_cast<const LineGap *>(detectorGap);
+
+        if (lineGap == nullptr)
+            continue;
+
+        // ATTN: Right now, this only works for gaps with only X components.
+        //       If the Z is used for detector gaps, can cause infs.
+        const float gapWidth = std::fabs(lineGap->GetLineStartX() - lineGap->GetLineEndX());
+
+        const float distanceToStart = std::fabs(testPoint.GetX() - lineGap->GetLineStartX());
+        const float distanceToEnd = std::fabs(testPoint.GetX() - lineGap->GetLineEndX());
+        const float middleOfGap = std::fabs((lineGap->GetLineStartX() + lineGap->GetLineEndX()) / 2);
+        const CartesianVector gapPoint(middleOfGap, 0, testPoint.GetZ());
+
+        const float distanceToGap = std::min(distanceToStart, distanceToEnd);
+        const float totalGapSize = distanceToGap + gapWidth;
```

Comment:
What's actually being computed here? This seems to be taking the minimum distance between a point inside the gap and the gap extrema, and adding it to the size of the gap, which seems to have some degree of double-counting.